### DISCUSSION TITLE
Keycloak login console keeps loading

### DISCUSSION THREAD

**Discussion by ajeetmind** [keycloak/keycloak#15017] [Q&A]
### Describe the bug

Hi Team ,

I am trying to run keycloak in aws fargate enviornmnet . when I click [Administration Console] from welcome page of keycloak.
Login page keeps loading. It seems this issue has been raised in past and team has fixed it but somwhow i am not able to figure out the solution . Error Screenshot
<img width="922" alt="image" src="https://user-images.githubusercontent.com/100093303/192843122-3c44e1ce-c5bb-4c0f-9ceb-a7babd294b7e.png">
Please find my fargate configuration details below . I am using dev mode with command start-dev. same is running fine in my docker but not on aws fargate.
{
  "ipcMode": null,
  "executionRoleArn": "arn:aws:iam::<id>:role/ecs-all-keycloak-acc",
  "containerDefinitions": [
    {
      "dnsSearchDomains": null,
      "environmentFiles": null,
      "logConfiguration": {
        "logDriver": "awslogs",
        "secretOptions": null,
        "options": {
          "awslogs-group": "/ecs/all-keycloak-acc",
          "awslogs-region": "eu-central-1",
          "awslogs-stream-prefix": "ecs"
        }
      },
      "entryPoint": null,
      "portMappings": [
        {
          "hostPort": 8080,
          "protocol": "tcp",
          "containerPort": 8080
        },
        {
          "hostPort": 8443,
          "protocol": "tcp",
          "containerPort": 8443
        }
      ],
      "command": [
        "start-dev"
      ],
      "linuxParameters": null,
      "cpu": 0,
      "environment": [
        {
          "name": " KC_PROXY",
          "value": "X-Forwarded-*"
        },
        {
          "name": "KC_DB",
          "value": "postgres"
        },
        {
          "name": "KC_DB_PASSWORD",
          "value": "<db passowrd>"
        },
        {
          "name": "KC_DB_URL",
          "value": "jdbc:postgresql://<aws-rds-postgres-url>:5432/fleet_keycloak_test"
        },
        {
          "name": "KC_DB_USERNAME",
          "value": "<db username>"
        },
        {
          "name": "KC_HEALTH_ENABLED",
          "value": "true"
        },
        {
          "name": "KC_HOSTNAME",
          "value": "<route 53 dns record>"
        },
        {
          "name": "KC_HOSTNAME_ADMIN",
          "value": "<route 53 dns record>"
        },
        {
          "name": "KC_HOSTNAME_STRICT_BACKCHANNEL",
          "value": "true"
        },
        {
          "name": "KC_HOSTNAME_STRICT_HTTPS",
          "value": "false"
        },
        {
          "name": "KEYCLOAK_ADMIN",
          "value": "admin"
        },
        {
          "name": "KEYCLOAK_ADMIN_PASSWORD",
          "value": "admin"
        },
        {
          "name": "PROXY_ADDRESS_FORWARDING",
          "value": "true"
        }
      ],
      "resourceRequirements": null,
      "ulimits": null,
      "dnsServers": null,
      "mountPoints": [],
      "workingDirectory": null,
      "secrets": null,
      "dockerSecurityOptions": null,
      "memory": null,
      "memoryReservation": null,
      "volumesFrom": [],
      "stopTimeout": null,
      "image": "quay.io/keycloak/keycloak:latest",
      "startTimeout": null,
      "firelensConfiguration": null,
      "dependsOn": null,
      "disableNetworking": null,
      "interactive": null,
      "healthCheck": null,
      "essential": true,
      "links": null,
      "hostname": null,
      "extraHosts": null,
      "pseudoTerminal": null,
      "user": null,
      "readonlyRootFilesystem": null,
      "dockerLabels": null,
      "systemControls": null,
      "privileged": null,
      "name": "keycloak"
    }
  ],
  "placementConstraints": [],
  "memory": "512",
  "taskRoleArn": "arn:aws:iam::<id>:role/ecs-all-keycloak-acc",
  "compatibilities": [
    "EC2",
    "FARGATE"
  ],
  "taskDefinitionArn": "arn:aws:ecs:eu-central-1:<id>:task-definition/all-keycloak-acc:5",
  "family": "all-keycloak-acc",
  "requiresAttributes": [
    {
      "targetId": null,
      "targetType": null,
      "value": null,
      "name": "com.amazonaws.ecs.capability.logging-driver.awslogs"
    },
    {
      "targetId": null,
      "targetType": null,
      "value": null,
      "name": "ecs.capability.execution-role-awslogs"
    },
    {
      "targetId": null,
      "targetType": null,
      "value": null,
      "name": "com.amazonaws.ecs.capability.docker-remote-api.1.19"
    },
    {
      "targetId": null,
      "targetType": null,
      "value": null,
      "name": "com.amazonaws.ecs.capability.task-iam-role"
    },
    {
      "targetId": null,
      "targetType": null,
      "value": null,
      "name": "com.amazonaws.ecs.capability.docker-remote-api.1.18"
    },
    {
      "targetId": null,
      "targetType": null,
      "value": null,
      "name": "ecs.capability.task-eni"
    }
  ],
  "pidMode": null,
  "requiresCompatibilities": [
    "FARGATE"
  ],
  "networkMode": "awsvpc",
  "runtimePlatform": {
    "operatingSystemFamily": "LINUX",
    "cpuArchitecture": null
  },
  "cpu": "256",
  "revision": 5,
  "status": "ACTIVE",
  "inferenceAccelerators": null,
  "proxyConfiguration": null,
  "volumes": []
}

Please help me to get login paage and running successfully.

### Version

quay.io/keycloak/keycloak:latest

### Expected behavior

 when I click [Administration Console] from welcome page of keycloak.
Login page keeps loading. It seems this issue has been raised in past and team has fixed it but somwhow i am not able to figure out the solution . Error Screenshot
<img width="922" alt="image" src="https://user-images.githubusercontent.com/100093303/192843122-3c44e1ce-c5bb-4c0f-9ceb-a7babd294b7e.png">
 I am using dev mode with command start-dev. same is running fine in my docker but not on aws fargate.

### Actual behavior

Logging page is keeps loading

### How to Reproduce?

Please use below mentioned json as task definiotion in AWS fargate platform and try to login
{
  "ipcMode": null,
  "executionRoleArn": "arn:aws:iam::<id>:role/ecs-all-keycloak-acc",
  "containerDefinitions": [
    {
      "dnsSearchDomains": null,
      "environmentFiles": null,
      "logConfiguration": {
        "logDriver": "awslogs",
        "secretOptions": null,
        "options": {
          "awslogs-group": "/ecs/all-keycloak-acc",
          "awslogs-region": "eu-central-1",
          "awslogs-stream-prefix": "ecs"
        }
      },
      "entryPoint": null,
      "portMappings": [
        {
          "hostPort": 8080,
          "protocol": "tcp",
          "containerPort": 8080
        },
        {
          "hostPort": 8443,
          "protocol": "tcp",
          "containerPort": 8443
        }
      ],
      "command": [
        "start-dev"
      ],
      "linuxParameters": null,
      "cpu": 0,
      "environment": [
        {
          "name": " KC_PROXY",
          "value": "X-Forwarded-*"
        },
        {
          "name": "KC_DB",
          "value": "postgres"
        },
        {
          "name": "KC_DB_PASSWORD",
          "value": "<db passowrd>"
        },
        {
          "name": "KC_DB_URL",
          "value": "jdbc:postgresql://<aws-rds-postgres-url>:5432/fleet_keycloak_test"
        },
        {
          "name": "KC_DB_USERNAME",
          "value": "<db username>"
        },
        {
          "name": "KC_HEALTH_ENABLED",
          "value": "true"
        },
        {
          "name": "KC_HOSTNAME",
          "value": "<route 53 dns record>"
        },
        {
          "name": "KC_HOSTNAME_ADMIN",
          "value": "<route 53 dns record>"
        },
        {
          "name": "KC_HOSTNAME_STRICT_BACKCHANNEL",
          "value": "true"
        },
        {
          "name": "KC_HOSTNAME_STRICT_HTTPS",
          "value": "false"
        },
        {
          "name": "KEYCLOAK_ADMIN",
          "value": "admin"
        },
        {
          "name": "KEYCLOAK_ADMIN_PASSWORD",
          "value": "admin"
        },
        {
          "name": "PROXY_ADDRESS_FORWARDING",
          "value": "true"
        }
      ],
      "resourceRequirements": null,
      "ulimits": null,
      "dnsServers": null,
      "mountPoints": [],
      "workingDirectory": null,
      "secrets": null,
      "dockerSecurityOptions": null,
      "memory": null,
      "memoryReservation": null,
      "volumesFrom": [],
      "stopTimeout": null,
      "image": "quay.io/keycloak/keycloak:latest",
      "startTimeout": null,
      "firelensConfiguration": null,
      "dependsOn": null,
      "disableNetworking": null,
      "interactive": null,
      "healthCheck": null,
      "essential": true,
      "links": null,
      "hostname": null,
      "extraHosts": null,
      "pseudoTerminal": null,
      "user": null,
      "readonlyRootFilesystem": null,
      "dockerLabels": null,
      "systemControls": null,
      "privileged": null,
      "name": "keycloak"
    }
  ],
  "placementConstraints": [],
  "memory": "512",
  "taskRoleArn": "arn:aws:iam::<id>:role/ecs-all-keycloak-acc",
  "compatibilities": [
    "EC2",
    "FARGATE"
  ],
  "taskDefinitionArn": "arn:aws:ecs:eu-central-1:<id>:task-definition/all-keycloak-acc:5",
  "family": "all-keycloak-acc",
  "requiresAttributes": [
    {
      "targetId": null,
      "targetType": null,
      "value": null,
      "name": "com.amazonaws.ecs.capability.logging-driver.awslogs"
    },
    {
      "targetId": null,
      "targetType": null,
      "value": null,
      "name": "ecs.capability.execution-role-awslogs"
    },
    {
      "targetId": null,
      "targetType": null,
      "value": null,
      "name": "com.amazonaws.ecs.capability.docker-remote-api.1.19"
    },
    {
      "targetId": null,
      "targetType": null,
      "value": null,
      "name": "com.amazonaws.ecs.capability.task-iam-role"
    },
    {
      "targetId": null,
      "targetType": null,
      "value": null,
      "name": "com.amazonaws.ecs.capability.docker-remote-api.1.18"
    },
    {
      "targetId": null,
      "targetType": null,
      "value": null,
      "name": "ecs.capability.task-eni"
    }
  ],
  "pidMode": null,
  "requiresCompatibilities": [
    "FARGATE"
  ],
  "networkMode": "awsvpc",
  "runtimePlatform": {
    "operatingSystemFamily": "LINUX",
    "cpuArchitecture": null
  },
  "cpu": "256",
  "revision": 5,
  "status": "ACTIVE",
  "inferenceAccelerators": null,
  "proxyConfiguration": null,
  "volumes": []
}

### Anything else?

Currently I have exposed it at above url https://acc.happygosh.com/

**Comment by stianst** [comment]
Firefox?

**Comment by ajeetmind** [comment]
I haven't tried in Firefox yet. I am facing issue in Edge and chrome

**Comment by stianst** [comment]
We believe this is fixed in https://github.com/keycloak/keycloak-ui/pull/3373

Could you please try the nightly release (quay.io/keycloak/keycloak:nightly) and check if that works?

**Comment by ajeetmind** [comment]
As mentioned, I deployed "image": "quay.io/keycloak/keycloak:nightly" now . But it is still having same issue.below is the URL
https://acc.happygosh.com/ .

**Comment by stianst** [comment]
If you take a look at the error message from the browser you'll see that it is complaining about including insecure content. Which is happening because KC_HOSTNAME_STRICT_HTTPS=false. Get rid of that and it should be fine.

**Comment by ajeetmind** [comment]
Now mentioned configuration is removed but issue is still same. My current configuration is mentioned below
KC_PROXY	X-Forwarded-*
KC_DB	postgres
KC_DB_PASSWORD	<DB password>
KC_DB_URL	<My DB URL>
KC_DB_USERNAME	fleet_keycloak_dev
KC_HEALTH_ENABLED	true
KC_HOSTNAME	acc.happygosh.com
KC_HOSTNAME_ADMIN	acc.happygosh.com
KC_HOSTNAME_STRICT_BACKCHANNEL	true
KEYCLOAK_ADMIN	admin
KEYCLOAK_ADMIN_PASSWORD	admin
PROXY_ADDRESS_FORWARDING	true

**Comment by stianst** [comment]
I'd suggest taking a look at https://www.keycloak.org/server/hostname

You've got an invalid value for KC_PROXY, and PROXY_ADDRESS_FORWARDING isn't an option at all.

Are you using start-dev? Or is this with start?

**Comment by ajeetmind** [comment]
As per your suggestion, I removed KC_PROXY, and PROXY_ADDRESS_FORWARDING and Still I am having same issue.
"environment": [
        {
          "name": "KC_DB",
          "value": "postgres"
        },
        {
          "name": "KC_DB_PASSWORD",
          "value": "<DB Password>"
        },
        {
          "name": "KC_DB_URL",
          "value": "<DB URL>"
        },
        {
          "name": "KC_DB_USERNAME",
          "value": "fleet_keycloak_dev"
        },
        {
          "name": "KC_HEALTH_ENABLED",
          "value": "true"
        },
        {
          "name": "KC_HOSTNAME",
          "value": "acc.happygosh.com"
        },
        {
          "name": "KC_HOSTNAME_ADMIN",
          "value": "acc.happygosh.com"
        },
        {
          "name": "KC_HOSTNAME_STRICT_BACKCHANNEL",
          "value": "true"
        },
        {
          "name": "KEYCLOAK_ADMIN",
          "value": "admin"
        },
        {
          "name": "KEYCLOAK_ADMIN_PASSWORD",
          "value": "admin"
        }
      ],

**Reply by saisiva249** [reply]
Did this issue solved, iam also facing the same issue, if any solution please let me know

**Comment by stianst** [comment]
I said invalid value for KC_PROXY, not to remove it. As you are clearly exposing Keycloak behind a proxy you need that.

Please check the docs https://www.keycloak.org/server/hostname and https://www.keycloak.org/server/reverseproxy. In the meantime I'm going to convert this to a discussion as I don't see a bug here.

**Reply by hemsoni** [reply]
Hi @stianst , Hi @ajeetmind , Did this work for you guys?
